include/nbl/builtin/hlsl/workgroup/shared_scan.hlsl

```diff
@@ -75,72 +73,59 @@ struct WorkgroupScanHead
             {
                 const uint prevLevelScan = sharedAccessor.main.get(gl_LocalInvocationIndex);
                 scan = subgroupOp(prevLevelScan);
-                if(isScan)
-                {
-                    sharedAccessor.main.set(scanStoreIndex, scan);
-                }
-            }
-            if(isScan)
-            {
-                scanStoreIndex += lastInvocationInLevel + 1u;
+                sharedAccessor.main.set(scanLoadIndex, scan);
             }
+            scanLoadIndex += lastInvocationInLevel + 1u;
```

Comment:
btw you can push this into the `if (participate)` if you push it into an `if` in the tail in the same way